### PR TITLE
refactor: Move Screens Config logic

### PR DESCRIPTION
**Pull request by cmaddox5**
Asana task: Investigate: [Add S3 operations to screens-config-lib repo](https://app.asana.com/0/1185117109217413/1205700210661299/f)

Moved all Config fetching/getting/putting from the screens repo. Logic is identical. Biggest difference is changing `SignsUi.S3Fetch` to implement the behaviour instead of having it's own standalone function.

### CODE REVIEW

File: lib/config/state/screens/s3_fetch.ex [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    put_operation = ExAws.S3.put_object(bucket, path, contents)
+
+    case ExAws.request(put_operation) do
+      {:ok, %{status_code: 200}} -> :ok
+      _ -> :error
+    end
+  end
+
+  defp config_path_for_environment do
+    "screens/screens-prod.json"
```
Comment (PaulJKim) [inline lib/config/state/screens/s3_fetch.ex:64]: Is this intended to be hardcoded for the prod file or should we parametize this to allow for fetching files for other environments?

Comment (cmaddox5) [reply lib/config/state/screens/s3_fetch.ex]: Good catch. I'm honestly not sure how this got there. I must have been testing something a while back and accidentally left it there. That is fixed now.

File: lib/config/state/screens/s3_fetch.ex [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      {:ok, Config.from_json(parsed), new_version}
+    else
+      :unchanged -> :unchanged
+      _ -> :error
+    end
+  end
+
+  @impl true
+  def get_config(current_version \\ nil) do
+    bucket = Application.get_env(:screens, :config_s3_bucket)
```
Comment (PaulJKim) [inline lib/config/state/screens/s3_fetch.ex:22]: A thought on all these calls to `Application.get_env/2` - does having it hardcoded to access the `:screens` config key mean that we need to add new config values to Screenplay with `:screens` as the root key in order to make this code work from there?

Comment (cmaddox5) [reply lib/config/state/screens/s3_fetch.ex:22]: It does yes. That brings up a good note that we should probably both change the config key to `:screens_config` and document that requirement in the README. I'll work on that now.

File: config/config.exs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import Config
+
+config :screens_config,
+  config_s3_bucket: "mbta-ctd-config"
+
+import_config "#{Mix.env()}.exs"
```
Comment (PaulJKim) [inline config/config.exs:6]: Nit: I don't know if this really makes a difference but I know that there's a preference for using `config_env()` over `Mix.env()` starting at Elixir 1.9 when `Mix.Config` was deprecated. Mentioned at the bottom of [this section of the docs](https://hexdocs.pm/elixir/1.12/Config.html#module-migrating-from-use-mix-config)

File: lib/config/state/screens/s3_fetch.ex [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    put_operation = ExAws.S3.put_object(bucket, path, contents)
+
+    case ExAws.request(put_operation) do
+      {:ok, %{status_code: 200}} -> :ok
+      _ -> :error
+    end
+  end
+
+  defp config_path_for_environment do
+    "screens/#{Application.get_env(:screens, :environment_name)}.json"
```
Comment (PaulJKim) [inline lib/config/state/screens/s3_fetch.ex:64]: So just to double check and be sure I'm following: This implies that in `screenplay`'s config, we'll need to add some new config values under the root key `:screens` that have the values `screens-dev`, `screens-dev-green`, and `screens-prod` depending on the environment, right?

**Review by PaulJKim — APPROVED** [review]
One minor comment and one clarifying question, but otherwise looks good!

**Comment by PaulJKim** [comment]
Ah whoops sorry, missed that CI was failing before approving

**Comment by cmaddox5** [comment]
We decided this is not worth doing. The convenience of the code being shared and the added complexity of mix config stuff kinda cancel each other out.

**Pull request closed without merging**